common.cmake

```diff
@@ -10,20 +10,32 @@ if(__NRFXLIB_COMMON_CMAKE__)
 endif()
 set(__NRFXLIB_COMMON_CMAKE__ TRUE)
 
+#
+# Build the path of the library taking in
+# -Arch type
+# -Float ABI
+# -wchar_t size
+#
 function(nrfxlib_calculate_lib_path lib_path)
+  # Add Arch type
+  assert(GCC_M_CPU "GCC_M_CPU must be set to find correct lib.")
+  set(temp_path lib/${GCC_M_CPU}/)
+  # Append floating ABI
   if(CONFIG_FLOAT)
 	if(CONFIG_FP_HARDABI)
       set(float_dir hard-float)
 	elseif(CONFIG_FP_SOFTABI)
       set(float_dir softfp-float)
 	else()
       assert(0 "Unreachable code")
-	endif()
+  endif()
   else()
 	set(float_dir soft-float)
   endif()
-
-  assert(GCC_M_CPU "GCC_M_CPU must be set to find correct lib.")
-
-  set(${lib_path} lib/${GCC_M_CPU}/${float_dir} PARENT_SCOPE)
+  string(APPEND temp_path "${float_dir}")
+  # Append short wchar_t (if enabled)
+  if (CONFIG_SHORT_WCHAR_T)
+    string(APPEND temp_path "/short-wchar")
+  endif()
+  set(${lib_path} "${temp_path}" PARENT_SCOPE)
```

Comment:
We want to keep the oneliner that describes the entire pattern.

`  set(${lib_path} lib/${GCC_M_CPU}/${float_dir}${wchar_size} PARENT_SCOPE)`

Re-write it to avoid string appending and retain this line.

Reply:
Please move comment to other PR

Reply:
Ah, I see it is already there. Deleting.